internal/controllers/products.go

```diff
@@ -46,8 +46,50 @@ func (c *Controller) CreateProduct(ctx context.Context, pInput ProductInput) err
 		Price:       pInput.Price,
 		Quantity:    pInput.Quantity,
 		AuthorID:    pInput.AuthorID,
-		CategoryID:  pInput.CategoryID,
+		CategoryID:  pCate.ID,
 	}
 
 	return c.Repository.CreateProduct(ctx, product)
 }
+
+// UpdateProduct updates a product in db given by product model in parameter
+func (c *Controller) UpdateProduct(ctx context.Context, id int, pInput ProductInput) error {
```

Comment:
```suggestion
func (c *Controller) UpdateProduct(ctx context.Context, pInput ProductInput) error {
```
Combine `id` into the ProductInput